Detections/SecurityEvent/GainCodeExecutionADFSViaSMB.yaml

```diff
@@ -0,0 +1,72 @@
+id: 12dcea64-bec2-41c9-9df2-9f28461b1295
+name: Gain Code Execution on ADFS Server via SMB + Remote Service/Scheduled Task
+description: |
+   'This query detects instances where an attacker has gained the ability to execute code on an ADFS Server through SMB and Remote Service/Scheduled Task.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: SecurityEvents
+    dataTypes:
+      - SecurityEvent
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - LateralMovement
+relevantTechniques:
+  - T1210
+tags:
+  - Solorigate
+query: |
+  let timeframe = 1d;
+  // Adjust for a longer timeframe for identifying ADFS Servers
+  let lookback = 6d;
+  // Identify ADFS Servers
+  let ADFS_Servers = (
+  SecurityEvent
+  | where TimeGenerated > ago(timeframe+lookback)
+  | where EventID == 4688 and SubjectLogonId != "0x3e4"
+  | where ProcessName has "Microsoft.IdentityServer.ServiceHost.exe"
+  | distinct Computer
+  );
+  SecurityEvent
+  | where TimeGenerated > ago(timeframe)
+  | where Computer in~ (ADFS_Servers)
+  | where Account !endswith "$"
+  // Check for SMB + remote service/scheduled task, event IDs for scheduled tasks
+  // Check for scheduled task events
+  | where EventID in (4697, 4698, 4699, 4700, 4701, 4702)
+  | extend d = parse_xml(EventData)
+  | extend SubjectLogonId = tostring(d.EventData.Data[3]["#text"])
+  // Check specifically for access to IPC$ share and PIPE\svcctl and PIPE\atsvc for Service Control Services and Schedule Control Services
+  | union ( 
+      SecurityEvent
+      | where TimeGenerated > ago(timeframe)
+      | where Computer in~ (ADFS_Servers)
+      | where Account !endswith "$"
+      | where EventID == 5145
+      | where RelativeTargetName == "svcctl" or RelativeTargetName  == "atsvc"
+  )
+  // Check for lateral movement
+  | join kind=inner
+  (SecurityEvent
+  | where TimeGenerated > ago(timeframe)
+  | where Account !endswith "$"
+      | where EventID == 4624 and LogonType == 3
+      | project SubjectLogonId = TargetLogonId 
+  ) on SubjectLogonId
+  | project TimeGenerated, Account, Computer, EventID, RelativeTargetName, IpAddress, IpPort
```

Comment:
I don't think you will get an IPAddress from the left side of this unless it shows up from the 5145, the others don't have an IP address field, for example - https://docs.microsoft.com/en-us/windows/security/threat-protection/auditing/event-4698.  Did you mean to bring thru the IPAddress from the 4624?

Reply:
Yes, I think I thought that the ones that had them could carry them over, but since it's not consistent that I'll take IP addresses off in the edit.